woeip/apps/air_quality/views.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import Error
+from django.shortcuts import redirect, render
+from django.utils.encoding import force_text
+
+from woeip.apps.air_quality import dustrak, forms, models
+
+logger = logging.getLogger(__name__)
+
+
+def index(request):
+    return render(request, 'index.html')
+
+
+@login_required
+def upload(request):
+    """Upload data for a session collected using the Dustrak air quality device and a separate GPS
+    log file. Creates a new Session instance, two SessionData instances, and Data instances for
+    each sample.
+    """
+
+    if request.method == 'POST':
+        form = forms.DustrakSessionForm(request.POST, request.FILES)
+        if form.is_valid():
+            form.save()
+            try:
+                air_sensor = models.Sensor.objects.get(name='Dustrak')
```

Comment:
What's going on here? Are sensors not specific to individual users? You're going to have more than one Dustrak or GPS sensor.

Reply:
This is actually a topic of discussion among the WOIEP team. You can see it at PR #40 and Issue #43 . I think it belongs in its own PR. In my opinion, this current PR is too large and unfocused already. I'd like to keep this sensor code for now, simply to get the agreed upon changes into the codebase.